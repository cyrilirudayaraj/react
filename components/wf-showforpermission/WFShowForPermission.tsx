import React from 'react';
import AuthUtil from '../../utils/AuthUtil';
import { connect } from 'react-redux';
import { getPermissions } from '../../slices/UserPermissionSlice';

interface WFShowForPermissionProps {
  permission?: any;
  userPermissions?: any;
  children?: any;
}
class WFShowForPermission extends React.Component<WFShowForPermissionProps> {
  render(): JSX.Element {
    const couldShow =
      this.props.permission &&
      !AuthUtil.isAuthorized(this.props.permission, this.props.userPermissions);
    let element = <React.Fragment></React.Fragment>;
    if (!couldShow) {
      element = <React.Fragment>{this.props.children}</React.Fragment>;
    }
    return element;
  }
}

const mapStateToProps = (state: any) => {
  return { userPermissions: getPermissions(state) };
};
export default connect(mapStateToProps)(WFShowForPermission);
