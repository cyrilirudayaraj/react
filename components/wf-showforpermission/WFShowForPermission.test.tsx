import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WFShowForPermission from './WFShowForPermission';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });
let wrapper: any;
let store: any;
beforeEach(() => {
  window['globalVars'] = {
    enablePermission: 1,
  };
  store = configureStore([])({
    userPermission: {
      userPermissions: ['task.update'],
    },
  });
  store.dispatch = jest.fn();
});

describe('snapshot test <WFShowForPermission>', () => {
  it('should render as expected', () => {
    wrapper = mount(
      <Provider store={store}>
        <WFShowForPermission permission={'task.update'} />
      </Provider>
    );
    wrapper.setProps({ userPermissions: ['task.update'] });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render as expected', () => {
    wrapper = mount(
      <Provider store={store}>
        <WFShowForPermission permission={'task.create'} />
      </Provider>
    );
    wrapper.setProps({ userPermissions: ['task.update'] });
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
