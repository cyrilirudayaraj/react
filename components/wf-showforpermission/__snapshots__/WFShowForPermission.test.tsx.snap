// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test <WFShowForPermission> should render as expected 1`] = `
"<Provider store={{...}} userPermissions={{...}}>
  <Connect(WFShowForPermission) permission=\\"task.update\\">
    <WFShowForPermission permission=\\"task.update\\" userPermissions={{...}} dispatch={[Function: mockConstructor]} />
  </Connect(WFShowForPermission)>
</Provider>"
`;

exports[`snapshot test <WFShowForPermission> should render as expected 2`] = `
"<Provider store={{...}} userPermissions={{...}}>
  <Connect(WFShowForPermission) permission=\\"task.create\\">
    <WFShowForPermission permission=\\"task.create\\" userPermissions={{...}} dispatch={[Function: mockConstructor]} />
  </Connect(WFShowForPermission)>
</Provider>"
`;
